jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.jbpm.services.task.assignment.impl;
+
+import org.jbpm.services.task.assignment.AssignmentService;
+import org.jbpm.services.task.assignment.AssignmentServiceRegistry;
+import org.kie.api.task.TaskContext;
+import org.kie.api.task.model.Status;
+import org.kie.api.task.model.Task;
+import org.kie.api.task.model.User;
+import org.kie.internal.task.api.TaskModelFactory;
+import org.kie.internal.task.api.TaskModelProvider;
+import org.kie.internal.task.api.assignment.Assignment;
+import org.kie.internal.task.api.assignment.AssignmentStrategy;
+import org.kie.internal.task.api.model.InternalTaskData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class AssignmentServiceImpl implements AssignmentService {
+    
+    private static final Logger logger = LoggerFactory.getLogger(AssignmentServiceImpl.class);
+    
+    private boolean enabled = Boolean.parseBoolean(System.getProperty("org.jbpm.task.assignment.enabled", "false"));
+
+    private AssignmentStrategy strategy;
+    private AssignmentServiceRegistry registry = AssignmentServiceRegistry.get();
+    private TaskModelFactory taskModelFactory = TaskModelProvider.getFactory();
+    
+    public AssignmentServiceImpl() {
+        this.strategy = registry.getStrategy(System.getProperty("org.jbpm.task.assignment.strategy", "PotentialOwnerBusyness"));        
+    }
+    
+    public AssignmentServiceImpl(AssignmentStrategy strategy) {        
+        this.strategy = strategy;
+    }
+    
+    @Override
+    public void assignTask(Task task, TaskContext context) {
+        
+        assignTask(task, context, null);
+    }    
+
+    @Override
+    public void assignTask(Task task, TaskContext context, String excludedUser) {
+        if (!isEnabled()) {
+            logger.debug("AssignmentService is not enabled - to enable it set system property 'org.jbpm.task.assignment.enabled' to true");
```

Comment:
The property name should be defined as a constant - it's been already used twice.